src/app/components/booking/index.tsx

```diff
@@ -166,17 +171,31 @@ export default function BookingComponent({
       return setOpenPrepaymentModal(true);
     }
 
+    if (totalPayment > 0) {
+      getCheckoutId()
+      return
+    }
     updateNotion(formData);
   };
 
+  const getCheckoutId = async () => {
+    const response = await generateCheckoutId(totalPayment.toString())
+    if (!response) {
+      return
+    }
+    setOpenPaymentModal(true)
+    setCheckoutId(response.id)
```

Comment:
We don't need to states here, we can check if the checkoutId has value in order to open the modal